### PR TITLE
feat: add software and software logs endpoints

### PR DESCRIPTION
**Pull request by bfabio**
See #11, #14.

### CODE REVIEW

File: internal/handlers/logs.go [hunk 1]
```diff
@@ -135,13 +139,96 @@ func (p *Log) PatchLog(ctx *fiber.Ctx) error {
...
+		}
+
+		return common.Error(
+			fiber.StatusInternalServerError,
+			"can't get Software",
+			fiber.ErrInternalServerError.Message,
+		)
+	}
+
+	stmt := p.db.Where("entity_id = ? AND entity_type = 'Software'", software.ID)
```
Comment (bfabio) [inline internal/handlers/logs.go:173]: I don't like this at all, but couldn't find a better way

File: internal/handlers/logs.go [hunk 2]
```diff
@@ -135,13 +139,96 @@ func (p *Log) PatchLog(ctx *fiber.Ctx) error {
...
+			"can't get Software",
+			fiber.ErrInternalServerError.Message,
+		)
+	}
+
+	log := models.Log{
+		ID:         utils.UUIDv4(),
+		Message:    logReq.Message,
+		EntityID:   software.ID,
+		EntityType: "Software",
```
Comment (bfabio) [inline internal/handlers/logs.go:226]: Same here

File: internal/handlers/logs.go (hunk 1 above)
Comment (LorenzoS92) [reply internal/handlers/logs.go:173]: You could it by struct & map conditions

https://gorm.io/docs/query.html#Struct-amp-Map-Conditions

File: internal/handlers/logs.go (hunk 2 above)
Comment (bfabio) [reply internal/handlers/logs.go:226]: That's what I didn't like: having to explicitly specify the type as a string, when we already have a `Software` struct and the ORM knows about the `Logs` <-> `Software` relation

File: internal/handlers/logs.go [hunk 3]
```diff
@@ -135,13 +139,98 @@ func (p *Log) PatchLog(ctx *fiber.Ctx) error {
...
+
+		return common.Error(
+			fiber.StatusInternalServerError,
+			"can't get Software",
+			fiber.ErrInternalServerError.Message,
+		)
+	}
+
+	stmt := p.db.
+		Where(map[string]interface{}{"entity_type": models.Software{}.TableName()}).
```
Comment (bfabio) [inline internal/handlers/logs.go:174]: @LorenzoS92 updated

Still ugly as sin, but at least it's not an hardcoded string

**Review by LorenzoS92 — DISMISSED** [review]

**Pull request merged**